storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestDefaultResourceAwareStrategy.java

```diff
@@ -104,79 +107,214 @@ public void cleanup() {
+                    assertThat(topologyResources.getAssignedNonSharedMemOffHeap(), closeTo(spoutParallelism * memoryOffHeap, 0.01));
+                    assertThat(numNodes, is(1L));
+                    assertThat(cluster.getAssignedNumWorkers(topo), is(spoutParallelism));
+                    break;
+                case sharedOffHeapWorkerType:
+                    // 4 workers on 2 nodes. OffHeapWorker memory not shared -- consumed 4x, once for each worker)
+                    assertThat(topologyResources.getAssignedMemOnHeap(), closeTo(spoutParallelism * memoryOnHeap, 0.01));
+                    assertThat(topologyResources.getAssignedMemOffHeap(), closeTo(spoutParallelism * (memoryOffHeap + sharedOffHeapWorker), 0.01));
+                    assertThat(topologyResources.getAssignedSharedMemOnHeap(), closeTo(0, 0.01));
+                    assertThat(topologyResources.getAssignedSharedMemOffHeap(), closeTo(spoutParallelism * sharedOffHeapWorker, 0.01));
+                    assertThat(topologyResources.getAssignedNonSharedMemOnHeap(), closeTo(spoutParallelism * memoryOnHeap, 0.01));
+                    assertThat(topologyResources.getAssignedNonSharedMemOffHeap(), closeTo(spoutParallelism * memoryOffHeap, 0.01));
+                    assertThat(numNodes, is(2L));
+                    assertThat(cluster.getAssignedNumWorkers(topo), is(spoutParallelism));
+                    break;
+                case sharedOnHeapType:
+                    // 4 workers on 2 nodes. onHeap memory not shared -- consumed 4x, once for each worker
+                    assertThat(topologyResources.getAssignedMemOnHeap(), closeTo(spoutParallelism * (memoryOnHeap + sharedOnHeap), 0.01));
+                    assertThat(topologyResources.getAssignedMemOffHeap(), closeTo(spoutParallelism * memoryOffHeap, 0.01));
+                    assertThat(topologyResources.getAssignedSharedMemOnHeap(), closeTo(spoutParallelism * sharedOnHeap, 0.01));
+                    assertThat(topologyResources.getAssignedSharedMemOffHeap(), closeTo(0, 0.01));
+                    assertThat(topologyResources.getAssignedNonSharedMemOnHeap(), closeTo(spoutParallelism * memoryOnHeap, 0.01));
+                    assertThat(topologyResources.getAssignedNonSharedMemOffHeap(), closeTo(spoutParallelism * memoryOffHeap, 0.01));
+                    assertThat(numNodes, is(2L));
+                    assertThat(cluster.getAssignedNumWorkers(topo), is(spoutParallelism));
+                    break;
+            }
+        }
+    }
 
-        scheduler = new ResourceAwareScheduler();
+    /**
+     * test if the scheduling shared memory is correct with/without oneWorkerPerExecutor enabled
+     */
+    @Test
+    public void testDefaultResourceAwareStrategySharedMemory() {
+        boolean oneWorkerPerExecutorValues[] = {false, true};
+        for (boolean oneWorkerPerExecutor : oneWorkerPerExecutorValues) {
```

Comment:
Same here, this should also be using `@ParameterizedTest`.

Reply:
again, would like to defer for now.

Reply:
test updated